pilot/pkg/status/resourcelock.go

```diff
@@ -14,47 +14,168 @@
 
 package status
 
```

Comment:
can you add
```
func TestMain(m *testing.M) {
	leak.CheckMain(m)
}
```

to the tests for this package? Should be useful for this since its a lot of low level concurrency/locking